lib/rouge/lexers/kotlin.rb

```diff
@@ -60,6 +64,21 @@ class Kotlin < RegexLexer
         end
         rule %r/\bfun\b/, Keyword
         rule /\b(?:#{keywords.join('|')})\b/, Keyword
+        rule %r'^\s*\[.*?\]', Name::Attribute
+        rule %r'[^\S\n]+', Text
+        rule %r'\\\n', Text # line continuation
+        rule %r'//.*?$', Comment::Single
+        rule %r'/[*].*?[*]/'m, Comment::Multiline
+        rule %r'\n', Text
+        rule %r'::|!!|\?[:.]', Operator
+        rule %r"(\.\.)", Operator
+        rule %r'[~!%^&*()+=|\[\]:;,.<>/?-]', Punctuation
+        rule %r'[{}]', Punctuation
+        rule %r'@"(""|[^"])*"'m, Str
+        rule %r'""".*?"""'m, Str
+        rule %r'"(\\\\|\\"|[^"\n])*["\n]'m, Str
+        rule %r"'\\.'|'[^\\]'", Str::Char
+        rule %r"[0-9](\.[0-9]+)?([eE][+-][0-9]+)?[flFL]?|0[xX][0-9a-fA-F]+[Ll]?", Num
```

Comment:
Is there something specific that this block represents? I know you simply moved it from above, but it would be helpful for future maintenance if this was at least commented to help understand what it does, or even to move it to a state for clarity.

Reply:
I moved the basic rules, for things like keywords, comments, text, numbers, strings etc. to be further down, so that more specialised rules would be matched first. As otherwise some specialised rules would be missed.

This rule for 'Num' was already there when I made my changes and it is there to match number constants. I would need to study the regex more to understand the regex completely, but I didn't change it, so it is as it was before anyway.